lib/services/kube_cluster_resource_wrapper.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"github.com/gravitational/teleport/api/types"
+)
+
+// KubeClusterResourceWrapper stubs out a minimal implementation of
+// services.AccessCheckable for *types.KubernetesCluster.
+// This allows us to use common RBAC checks for Kubernetes Clusters
+// and other resources.
```

Comment:
The only thing really missing from `KubernetesCluster` is namespace, right?
How about implementing the missing methods (except `GetMetadata`) on `KubernetesCluster` itself, and embedding it in this wrapper. The wrapper can basically be:
```
type KubeClusterResourceWrapper {
  *types.KubernetesCluster
  Namespace string
}
func (k KubeClusterResourceWrapper) GetMetadata() types.Metadata {
	return types.Metadata{
		Name:      k.GetName(),
		Namespace: types.ProcessNamespace(k.Namespace),
	}
}
```

Reply:
That works, I was afraid of making `KubernetesCluster` look too much like a `types.Resource` and making this a much larger refactor, but I like this suggestion.

Reply:
I've applied your suggestion to just take the namespace instead of holding a reference to the server, but am still not sold on modifying the `api/types` package. I don't want to confuse people consuming the API by making this look like any other resource when it's actually handled quite differently.

What do you think @awly? Good as is or would you really want to see this change?